core/storage/btree.rs

```diff
@@ -358,16 +367,87 @@ impl BTreeCursor {
     }
 
     fn move_to_root(&mut self) {
-        self.page
-            .replace(Some(Rc::new(MemPage::new(None, self.root_page, 0))));
+        let mem_page = self.pager.read_page(self.root_page).unwrap();
+        self.stack.borrow_mut()[0] = Some(mem_page);
+        self.cell_indices.borrow_mut()[0] = 0;
+        *self.current_page.borrow_mut() = 0;
+    }
+
+    fn push_to_stack(&self, page: Rc<RefCell<Page>>) {
```

Comment:
could we name these `stack_push()`, `stack_pop()`, `stack_peek()`?

Reply:
maybe I will extract stack to other place, maybe not who knows